infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/cache/jpa/JpaKubernetesRuntimeStateCache.java

```diff
@@ -179,7 +179,7 @@ public void remove(RuntimeIdentity runtimeId) throws InfrastructureException {
     }
   }
 
-  @Transactional
+  @Transactional(rollbackOn = {RuntimeException.class, ServerException.class})
```

Comment:
I think `InfrastructureException` is not a correct pattern here. Since ServerException comes from `eventService.X..propagateException();` better to follow this pattern  https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/jpa/JpaWorkspaceDao.java#L236-L248